HelpSource/Guides/WritingClasses.schelp

```diff
@@ -106,101 +186,176 @@ we are calling the code::.init:: method on that object, which is an instance met
 
-section:: Overriding Methods (Overloading)
+code::
+MyClass {
+	classvar myClassvar;
 
-In order to change the behaviour of the superclass, often methods are overridden. Note that an object looks always for
-the method it has defined first and then looks in the superclass. Here code::MyClass.value(2):: will return 6, not 4:
+	*initClass {
+		myClassvar = IdentityDictionary.new;
+	}
+}
+::
+
+
+
+
+section::Alternatives to inheritance
+
+Overreliance on inheritance is usually a design flaw. Inheritance is mainly a way to organise code, and shouldn't be mistaken for a categorisation of objects. Two objects may respond to a message in different ways (polymorphism), and objects delegate control to ther objects they hold in their instance variables (object composition).
+
+subsection::Polymorphism
+
+See also: link::Guides/Polymorphism::
+
+Two completely unrelated objects can respond to the same messages and therefore be used together in the same code. For example, link::Classes/Function:: and link::Classes/Event:: have no common superclass apart from the general class link::Classes/Object::. But both respond to the message code::play::. Instead of inheriting all methods, you can simply implement some of the same methods in your class.
 
 code::
-Superclass {
-    calculate { arg in; in * 2 }
-    value { arg in; ^this.calculate(in) }
+MyClass {
+	var count = 0;
+	value {
+		^count = count + 1
+	}
 }
+// objects of this class will respond to the message "value", just like a function.
+a = MyClass.new;
+a.value; // returns 1
+::
+
 
-MyClass : Superclass {
-    calculate { arg in; in * 3 }
+subsection::Object Composition
+Often, an object passes control to one of the objects it has in its instance variables. Because these objects can be of any kind, this is a very flexible way to achieve a wide range of functionalities. For example, a link::Classes/Button:: has an code::action:: instance variable, which may hold anything that responds to the message code::value::.
+
+code::
+MyClass {
+	var action;
+	*new { |action|
+		^super.newCopyArgs(action)
+	}
+	value {
+		action.value;
+	}
 }
+
+// depending on what "action" is, objects of this class will behave differently
+a = MyClass({ "hello." });
+b = MyClass({ |i| log2(i) * sin(i * pi) });
+a.value(8);
+b.value(8);
 ::
 
-If the method of the superclass is needed, it can be called by super.
+Often, variables like code::action:: above are filled with custom objects that belong to code::MyClass::. Thus, one will write many small classes that can be well combined in such a way. This is called "pluggable behavior".
 
-code::
-Superclass {
-    var x;
 
-    init {
-        x = 5;
-    }
+section::Variables
+
+subsection:: Initializing variables directly
+
+In a variable declaration, variables can be directly initialized. This works for link::Reference/Literals:: only.
```

Comment:
The second sentence would be clearer as "Only literals may be used to initialize variables this way."